docs/getting/tutorial.rst

```diff
@@ -0,0 +1,88 @@
+.. _tutorial:
+
+**************************
+Tutorial
+**************************
+
+This example will show the simplist way to use pandas with pint and the underlying objects.
+It's slightly fiddly to set up units compared to reading data and units from a file.
+A more typical use case is given in :doc:`Reading from csv <../user/reading>`.
+
+
+Imports
+-----------------------
+First some imports
+
+.. ipython:: python
+
+   import pandas as pd
+   import pint
+   import pint_pandas
+
+   pint_pandas.show_versions()
+
+
+Create a DataFrame
+-----------------------
+Next, we create a DataFrame with PintArrays as columns.
+
+.. ipython:: python
+
+   df = pd.DataFrame(
+      {
+         "torque": pd.Series([1.0, 2.0, 2.0, 3.0], dtype="pint[lbf ft]"),
+         "angular_velocity": pd.Series([1.0, 2.0, 2.0, 3.0], dtype="pint[rpm]"),
+      }
+   )
+   df
+
+
+DataFrame Operations
+-----------------------
+Operations with columns are units aware so behave as we would intuitively expect.
+
+.. ipython:: python
+
+   df["power"] = df["torque"] * df["angular_velocity"]
+   df
+
+
+.. note::
+
+   Notice that the units are not displayed in the cells of the DataFrame.
+   If you ever see units in the cells of the DataFrame, something isn't right.
+   See :ref:`units_in_cells` for more information.
+
+We can see the columns' units in the dtypes attribute
+
+.. ipython:: python
+
+   df.dtypes
+
+Each column can be accessed as a Pandas Series
+
+.. ipython:: python
+
+   df.power
+
+Which contains a PintArray
+
+.. ipython:: python
+
+   df.power.values
+
+The PintArray contains a Quantity
+
+.. ipython:: python
+
+   df.power.values.quantity
+
+Pandas Series Accessors
+-----------------------
+Pandas Series accessors are provided for most Quantity properties and methods.
+Methods that return arrays will be converted to Series.
+
+.. ipython:: python
+
+   df.power.pint.units
+   df.power.pint.to("kW")
```

Comment:
Add:

Pandas and PintArrays
----------------------
Pandas makes it easy to put data into and rows and columns, and most novice users do not need to understand any details beyond the fact that a DataFrame has both rows and columns, whereas Series are 1-dimensional arrays (which could be a row of data or a column of data).  But when using advanced features of Pandas like ExtensionArrays, it is helpful to understand a few additional details.

When working with basic numerical data, Pandas uses Numpy data structures which are well-suited to vectorization and other performance optimizations.  Pandas ExtensionArrays provide almost the full range of Pandas functionality when operating on user-defined 1-dimensional arrays.  A PintArray is an ExtensionArray that is filled with Pint Quantities (and which can be optimized for performance).

A 1-dimensional Pandas Series can use a PintArray to hold its values.  Columns in 2-dimensional Pandas DataFrame can contain PintArrays--with all the efficiency the ExtensionArray APIs provide, but rows are a special case.  If all elements of the row have the same units, the row will be returned as Series backed by a PintArray with those units.  But if the units are heterogeneous, the row will be returned as a Series consisting of discrete Quantities (or raw data if the column values don't have units).  All Quantity data within such Series will follow Pint rules of unit conversions and will give error messages when units are not compatible, but some error messages may lose information as Pandas tries to align two incompatible Quantities to non-unitized magnitude values.  To get the greatest benefit from Pint-pandas (and Pandas in general), make your columns from data with homogeneous data and let your rows contain the heterogeneous data when necessary.

When examining DataFrames that contain units, if you see units within your DataFrame or Series, it means the Pandas object is not using PintArrays but it is still using Pint Quantities.  If you see only magnitudes when you print your DataFrame, and you see `pint[units]` as the `dtype` of the column or Series, it means the Pandas object is using PintArrays.

Reply:
> df.power.pint.m_as("kW")  # convert to a magnitude for specific numeric operation purposes
I think this could be a link to its own page for all series/dataframe accessors. that can be another PR

Reply:
> Pandas and PintArrays
again this is too much for the tutorial and should be on another page/in common issues